components/PicklistBase/PicklistBase.vue

```diff
@@ -576,32 +591,50 @@ export default {
        */
       this.$emit('valid', this.isValid);
     },
-    picked: {
-      handler() {
-        this.pickedRows = this.picked;
+    pickedAsArray: {
+      handler(newPickedEntries) {
+        this.syncPickedRows(newPickedEntries);
       },
-      deep: true,
+      deep: true, // Ensure the watcher goes deep into the structure
     },
     pickedRows: {
       handler() {
-        /**
-         * There has been a change to the picked rows.
-         *
-         * @property {Array} pickedRows Index `i` indicates the state of row `i`. Row `i` will be `1` or a non-zero number (if using a `quantityAttribute`) if selected.  Row `i` will be `0` if not selected.
-         */
-        this.$emit('update:picked', this.pickedRows);
+        const pickedMap = new Map();
+        for (let i = 0; i < this.pickedRows.length; i++) {
+          if (this.pickedRows[i] > 0) {
+            // Only add non-zero selections
+            const row = this.sortedRows[i];
+            const originalIndex = this.rows.indexOf(row);
+            if (originalIndex !== -1) {
+              pickedMap.set(originalIndex, {
+                row,
+                quantity: this.pickedRows[i],
```

Comment:
Let's change this property to be `picked`.  The `row` already contains an entry for `quantity`